industrial_ci/src/tests/ros_prerelease.sh

```diff
@@ -72,41 +73,26 @@ function run_ros_prerelease() {
 
     ici_time_start setup_prerelease_scripts
     mkdir -p "$WORKSPACE/catkin_workspace/src/"
-    local reponame=$PRERELEASE_REPONAME
-    local clone_underlay=true
-    if [ -z "$reponame" ]; then
-        reponame=$TARGET_REPO_NAME
-        mkdir -p catkin_workspace/src
-        cp -a "$TARGET_REPO_PATH"/* "$WORKSPACE/catkin_workspace/src/"
-        clone_underlay=false
-    fi
+    local reponame=${PRERELEASE_REPONAME:-$TARGET_REPO_NAME}
+    cp -a "$TARGET_REPO_PATH" "$WORKSPACE/catkin_workspace/src/$reponame"
 
     # ensure access rights
     ici_run_cmd_in_docker $DIND_OPTS -v "$WORKSPACE:$WORKSPACE:rw"  --user root  "industrial-ci/prerelease" chown -R ci:ci $WORKSPACE
 
-    run_in_prerelease_docker generate_prerelease_script.py https://raw.githubusercontent.com/ros-infrastructure/ros_buildfarm_config/production/index.yaml "$ROS_DISTRO" default ubuntu "$UBUNTU_OS_CODE_NAME" amd64 "${reponame}" --level "$downstream_depth" --output-dir . 
-    ici_time_end  # setup_prerelease_scripts
 
-    if [ "$clone_underlay" = true ]; then
-        ici_time_start prerelease_clone_underlay.sh
-        run_in_prerelease_docker ./prerelease_clone_underlay.sh
-        ici_time_end  # prerelease_clone_underlay
+    if [ "${USE_MOCKUP// }" != "" ]; then
+        if [ ! -d "$TARGET_REPO_PATH/$USE_MOCKUP" ]; then
+            error "mockup directory '$USE_MOCKUP' does not exist"
+        fi
+        cp -a "$TARGET_REPO_PATH/$USE_MOCKUP" "$WORKSPACE/catkin_workspace/src"
     fi
 
-    ici_time_start prerelease_build_underlay.sh
-    run_in_prerelease_docker ./prerelease_build_underlay.sh
-    run_in_prerelease_docker catkin_test_results $WORKSPACE/catkin_workspace/test_results
-    ici_time_end  # prerelease_build_underlay.sh
-
-    if [ "$downstream_depth" != "0" ]; then
-        ici_time_start run_prerelease_clone_overlay.sh
-        run_in_prerelease_docker ./prerelease_clone_overlay.sh
-        ici_time_end  # run_prerelease_clone_overlay
-        ici_time_start prerelease_build_overlay.sh
-        run_in_prerelease_docker ./prerelease_build_overlay.sh
-        run_in_prerelease_docker catkin_test_results $WORKSPACE/catkin_workspace_overlay/test_results
-        ici_time_end  # prerelease_build_overlay.sh
-    fi
+    run_in_prerelease_docker generate_prerelease_script.py https://raw.githubusercontent.com/ros-infrastructure/ros_buildfarm_config/production/index.yaml "$ROS_DISTRO" default ubuntu "$UBUNTU_OS_CODE_NAME" amd64 --level "$downstream_depth" --output-dir . --custom-repo "$reponame::::"
```

Comment:
Yes, all of them.

repo **:** type **:** url_scheme **:** url_path **:** branch
For us type, url_scheme, url_patch, branch are empty.
This way the prerelease will fail if the target source code is missing.